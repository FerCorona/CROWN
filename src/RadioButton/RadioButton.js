import React from 'react';

import Label from '../Label';

const RadioButton = ({
  options,
  optionSelected,
  onChange,
  stateName,
  extraStyle
}) => (
  <div className={`CROWNRadioButton ${extraStyle}`} role='radiogroup' aria-labelledby='bulgy-radios-label'>
    {
      options.map((option, key) => (
        <label key={key}>
          <input
            type='radio'
            value={option.id}
            checked={optionSelected === option.id}
            onChange={() => onChange({
              stateName,
              value: option.id
            })}
          />
          <span className='Radio' />
          <div className='Label'>
            <Label size='small' text={option.label} color='--black-1' />
          </div>
        </label>
      ))
    }
  </div>
);

RadioButton.defaultProps = {
  options: [
    {
      label: 'Si',
      id: 'si_option'
    },
    {
      label: 'No',
      id: 'no_option'
    },
    {
      label: 'Tal vez',
      id: 'tal_vez_option'
    }
  ],
  onChange: isChecked => console.log(isChecked),
  extraStyle: ''
};

export default RadioButton;