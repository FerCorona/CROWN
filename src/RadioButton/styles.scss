
.CROWNRadioButton {
  margin-left: 5px;
  .Label {
    margin-left: 10px;
  }
  label {
    display: block;
    position: relative;
    height: 0.5em;
    padding-left: 1rem;
    cursor: pointer;
    font-size: 2.5rem;
    user-select: none;
    color: #555;
    letter-spacing: 1px;
    &:hover input:not(:checked) ~ .Radio {
      opacity: 0.8;
    }
  }
  input {
    position: absolute;
    cursor: pointer;
    height: 0;
    width: 0;
    left: -2000px;
    &:checked {
      ~ .Radio {
        background-color: #00bcd4;
        transition: background .3s;
        &::after {
          opacity: 1;
        }
      }
    }
  }
  .Radio {
    position: absolute;
    top: 0.2rem;
    left: 0;
    height: 1rem;
    width: 1rem;
    background: #c9ded6;
    border-radius: 50%;
    &::after {
      content: '';
      position: absolute;
      opacity: 0;
      top: .25rem;
      left: .25rem;
      width: .5rem;
      height: .5rem;
      border-radius: 50%;
      background: #fff;
    }
  }
}
