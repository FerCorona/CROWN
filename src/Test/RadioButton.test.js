import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from '../RadioButton/index';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <RadioButton />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});