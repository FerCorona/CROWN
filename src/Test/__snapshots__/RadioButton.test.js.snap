// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  aria-labelledby="bulgy-radios-label"
  className="CROWNRadioButton "
  role="radiogroup"
>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="si_option"
    />
    <span
      className="Radio"
    />
    <div
      className="Label"
    >
      <div
        className="CROWNLabel "
        style={
          Object {
            "color": "var(--black-1)",
            "fontSize": "small",
            "fontWeight": "400",
          }
        }
      >
        Si
      </div>
    </div>
  </label>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="no_option"
    />
    <span
      className="Radio"
    />
    <div
      className="Label"
    >
      <div
        className="CROWNLabel "
        style={
          Object {
            "color": "var(--black-1)",
            "fontSize": "small",
            "fontWeight": "400",
          }
        }
      >
        No
      </div>
    </div>
  </label>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="radio"
      value="tal_vez_option"
    />
    <span
      className="Radio"
    />
    <div
      className="Label"
    >
      <div
        className="CROWNLabel "
        style={
          Object {
            "color": "var(--black-1)",
            "fontSize": "small",
            "fontWeight": "400",
          }
        }
      >
        Tal vez
      </div>
    </div>
  </label>
</div>
`;
